x-pack/plugins/security_solution/public/common/lib/upsellings/upselling_service.ts

```diff
@@ -11,12 +11,12 @@ import type { SecurityPageName } from '../../../../common';
 import type { SectionUpsellings, PageUpsellings, UpsellingSectionId } from './types';
 
 export class UpsellingService {
-  private sections: Map<UpsellingSectionId, React.ComponentType>;
+  private sections: Map<UpsellingSectionId, React.ComponentType | string>;
```

Comment:
Have we considered creating a separate structure for _string_ upselling messages? Like
`private messages: Map<UpsellingSectionId, string>;`
We'd need an additional `registerMessages`, but we won't mix types and it will be easier to consume. 
wdyt?

Reply:
Potential improvement: 
If we want to avoid re-renders when using `useUpsellingComponent` hook, we can export a new set of methods from this class, for sections would be something like:
```
  getSectionsValue() {
    return this.sectionsSubject$.getValue();
  }
```
And use them to set the initial value of the `useObservable` in the upselling hook:
```
  const upsellingSections = useObservable(upselling.sections$, upselling.getSectionsValue());
```
So the initial value will be the latest value instead of `undefined`, and we prevent a component re-render when using it.